artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/AMQPMessageSupport.java

```diff
@@ -396,36 +371,54 @@ private static CoreMessage newMessage(long id, byte messageType, CoreMessageObje
    }
 
 
-   public static byte destinationType(Destination destination) {
-      if (destination instanceof Queue) {
-         if (destination instanceof TemporaryQueue) {
-            return TEMP_QUEUE_TYPE;
-         } else {
-            return QUEUE_TYPE;
-         }
-      } else if (destination instanceof Topic) {
-         if (destination instanceof TemporaryTopic) {
-            return TEMP_TOPIC_TYPE;
-         } else {
-            return TOPIC_TYPE;
-         }
+   public static String toAddress(String destination) {
+
+      if (destination.startsWith(DestinationUtil.QUEUE_QUALIFIED_PREFIX)) {
+         return destination.substring(DestinationUtil.QUEUE_QUALIFIED_PREFIX.length());
+      } else if (destination.startsWith(DestinationUtil.TOPIC_QUALIFIED_PREFIX)) {
+         return destination.substring(DestinationUtil.TOPIC_QUALIFIED_PREFIX.length());
+      } else if (destination.startsWith(DestinationUtil.TEMP_QUEUE_QUALIFED_PREFIX)) {
+         return destination.substring(DestinationUtil.TEMP_QUEUE_QUALIFED_PREFIX.length());
+      } else if (destination.startsWith(DestinationUtil.TEMP_TOPIC_QUALIFED_PREFIX)) {
+         return destination.substring(DestinationUtil.TEMP_TOPIC_QUALIFED_PREFIX.length());
+      } else {
+         return destination;
       }
+   }
 
+   public static byte destinationType(String destination) {
+      if (destination.startsWith(QUEUE_QUALIFIED_PREFIX)) {
+         return QUEUE_TYPE;
+      }
+      if (destination.startsWith(TOPIC_QUALIFIED_PREFIX)) {
+         return TOPIC_TYPE;
+      }
+      if (destination.startsWith(TEMP_QUEUE_QUALIFED_PREFIX)) {
+         return TEMP_QUEUE_TYPE;
+      }
+      if (destination.startsWith(TEMP_TOPIC_QUALIFED_PREFIX)) {
+         return TEMP_TOPIC_TYPE;
+      }
       return QUEUE_TYPE;
    }
 
-   public static Destination destination(byte destinationType, String address) {
+   public static String destination(RoutingType destinationType, String address) {
+      String prefix;
+
+      if (destinationType == null) {
+         destinationType = RoutingType.ANYCAST;
+      }
+
       switch (destinationType) {
-         case TEMP_QUEUE_TYPE:
-            return new ActiveMQTemporaryQueue(address, null);
-         case TEMP_TOPIC_TYPE:
-            return new ActiveMQTemporaryTopic(address, null);
-         case TOPIC_TYPE:
-            return new ActiveMQTopic(address);
-         case QUEUE_TYPE:
-            return new ActiveMQQueue(address);
-         default:
-            return new ActiveMQQueue(address);
+         case ANYCAST: prefix = QUEUE_QUALIFIED_PREFIX; break;
```

Comment:
This looks like it no longer handles temporaries

Reply:
The method that uses routing type has the same semantics here.

Reply:
The original method was creating a Queue type (TemporaryQueue, TemporaryTopic, Queue, Topic), to set on the JMS Object, to then only use the simple name inside the object.

The Queue type was being set somewhere else, and this was not being used.

This method now is being use to replace ActiveMQDEstiation.createQueue

There is a test that was already validating this and it is passing.